tests/integration_python/test_run_cli.py

```diff
@@ -274,3 +276,43 @@ def test_run_with_activation(pixi: Path, tmp_pixi_workspace: Path) -> None:
         [pixi, "run", "--manifest-path", manifest, "--force-activate", "task", "-vvv"],
         stdout_contains="test123",
     )
+
+
+def test_detached_environments_run(pixi: Path, tmp_path: Path, dummy_channel_1: str) -> None:
+    manifest = tmp_path.joinpath("pixi.toml")
+
+    # Create a dummy project
+    verify_cli_command([pixi, "init", tmp_path, "--channel", dummy_channel_1])
+    verify_cli_command([pixi, "add", "dummy-a", "--no-install", "--manifest-path", manifest])
+
+    # Set detached environments
+    verify_cli_command(
+        [
+            pixi,
+            "config",
+            "set",
+            "--manifest-path",
+            manifest,
+            "--local",
+            "detached-environments",
+            "/tmp/pixi-detached-envs",
+        ],
+    )
+
+    # Run the installation
+    verify_cli_command([pixi, "install", "--manifest-path", manifest])
+
+    # Validate the detached environment
+    detached_envs = Path("/tmp/pixi-detached-envs")
+    assert detached_envs.exists()
+    detached_envs_folder = next(
+        (folder for folder in detached_envs.iterdir() if folder.is_dir()), "no_folder"
+    )
+    # Validate the conda-meta folder exists
+    assert Path(detached_envs_folder).joinpath("envs", "default", "conda-meta").exists()
```

Comment:
That is maybe a bit too clever :)

```suggestion
    detached_envs_folder = None
    for folder in detached_envs.iterdir():
        if folder.is_dir():
            detached_envs_folder = folder
            break

    assert detached_envs_folder is not None, "Couldn't find detached envs folder"

    # Validate the conda-meta folder exists
    assert detached_envs_folder.joinpath("envs", "default", "conda-meta").exists()
```